src/Resources/app/storefront/src/js/filter-category-select.plugin.js

```diff
@@ -105,6 +120,98 @@ export default class FilterCategorySelectPlugin extends FilterBasePlugin {
         this.listing.changeListing();
     }
 
+    /**
+     * @public
+     */
+    subCategoryDisplay(checkbox) {
+        let inner_sub_cats = checkbox.parentNode.getElementsByClassName('sub-item');
+        let span = checkbox.parentNode.querySelector('#arrow');
+        
+        if(checkbox.checked) {
+            let split_cats = checkbox.value.split('_');
+            if(split_cats.length > 0) {
+                Iterator.iterate(split_cats,(id) => {
+                    document.getElementById(id).checked = true;
+                })
+            }
+            this.toggleCategoryArrows(span,'down-arrow','up-arrow');
+            this.toggleSubcategoryDisplay(inner_sub_cats,'subcats-hide','subcats-show');
+        } else {
+            Iterator.iterate(inner_sub_cats, (sub_cat)=> {
+                sub_cat.querySelector('.filter-category-select-checkbox').checked = false;
+            });
+        }
+    }
+
+    /**
+     * @public
+     */
+    toggleCategoryArrows(elem,removeClass,addClass) {
```

Comment:
You are not toggling the arrows, you are toggling the icon state of the arrows, so a method name such as `toggleCategoryIconState`, would be more sophisticated.

Reply:
This wasn't taken care of.